src/System.Net.Security/tests/FunctionalTests/CertificateValidationClientServer.cs

```diff
@@ -104,11 +120,15 @@ public void Dispose()
                             TestConfiguration.PassingTestTimeoutMilliseconds),
                         "Client/Server Authentication timed out.");
 
-                    Assert.True(sslClientStream.IsMutuallyAuthenticated, "sslClientStream.IsMutuallyAuthenticated");
-                    Assert.True(sslServerStream.IsMutuallyAuthenticated, "sslServerStream.IsMutuallyAuthenticated");
+                    if (!_clientCertificateRemovedByFilter)
+                    {
+                        Assert.True(sslClientStream.IsMutuallyAuthenticated, "sslClientStream.IsMutuallyAuthenticated");
+                        Assert.True(sslServerStream.IsMutuallyAuthenticated, "sslServerStream.IsMutuallyAuthenticated");
+
+                        Assert.Equal(sslServerStream.RemoteCertificate.Subject, _clientCertificate.Subject);
+                    }
```

Comment:
I agree: it would make the behavior easier to understand.
I actually wanted to somehow skip the test but couldn't find a way to achieve that with `ConditionalTheory`.